network-api/networkapi/wagtailpages/pagemodels/research_hub/detail_page.py

```diff
@@ -180,21 +186,26 @@ def __str__(self):
 
     def clean(self):
         super().clean()
-        if self.url and self.document:
-            error_message = "Please provide either a URL or a document, not both."
+
+        # Ensure that only one of the three fields is set
+        if sum([bool(self.url), bool(self.page), bool(self.document)]) > 1:
+            error_message = "Please provide either a URL, a page or a document, not multiple."
             raise exceptions.ValidationError(
-                {"url": error_message, "document": error_message},
+                {"url": error_message, "page": error_message, "document": error_message},
                 code="invalid",
             )
-        elif not self.url and not self.document:
-            error_message = "Please provide a URL or a document."
+        # Ensure that at least one of the three fields is set
+        elif not any([self.url, self.page, self.document]):
+            error_message = "Please provide a URL, a page or a document."
             raise exceptions.ValidationError(
-                {"url": error_message, "document": error_message},
+                {"url": error_message, "page": error_message, "document": error_message},
                 code="required",
             )
 
     def get_url(self):
         if self.url:
             return self.url
+        elif self.page:
+            return self.page.get_url()
```

Comment:
**Potential issue**

I'm sure how much of a priority this edge case is but due to the nature of streamfield data storing only references, we probably want to add a check for `and self.page.live` here too. Because if an unpublished page is still referenced it will 404 for the user. Along with this, we will need to add an `if not get_url` check in the template for this link. Otherwise the link will render as `None`
Deleting a referenced page looks fine.

Reply:
Thanks for catching this!

Reply:
I wonder if we should prevent linking to non-live pages... ah... but a page can become not-live after it has been linked to! 

Handling a non-live page it is.

Reply:
Implemented the check and am not rendering the link button for non-live pages.